### PR TITLE
Replace deprecated react-helmet with thread-safe react-helmet-async

### PR DESCRIPTION
**Pull request by wlindner**
- https://open.nytimes.com/the-future-of-meta-tag-management-for-modern-react-development-ec26a7dc9183

## Description

React-helmet is no longer maintained and has been forked and replaced by react-helmet-async. This PR fixes this for the Gatsby starter projects.

This is the description of the issue: https://open.nytimes.com/the-future-of-meta-tag-management-for-modern-react-development-ec26a7dc9183

### Documentation

https://github.com/gatsbyjs/gatsby/tree/master/starters

## Related Issues

Related to https://github.com/nfl/react-helmet/pull/472
Deprecation explained here https://github.com/nfl/react-helmet/issues/426

### CODE REVIEW

File: starters/blog/src/components/seo.js [hunk 1]
```diff
@@ -28,47 +28,50 @@ function SEO({ description, lang, meta, title }) {
...
-          name: `twitter:title`,
-          content: title,
-        },
-        {
-          name: `twitter:description`,
-          content: metaDescription,
-        },
-      ].concat(meta)}
-    />
+    <HelmetProvider>
```
Comment (lannonbr) [inline starters/blog/src/components/seo.js]: As a notice, `gatsby-plugin-react-helmet-async` has a [gatsby-browser](https://github.com/Collabsalot/gatsby-plugin-react-helmet-async/blob/master/src/gatsby-browser.tsx) / [gatsby-ssr](https://github.com/Collabsalot/gatsby-plugin-react-helmet-async/blob/master/src/gatsby-ssr.tsx) script that adds this HelmetProvider with wrapRootElement so I believe this won't be needed in these component files, but it is just added as part of the gatsby plugin itself.

Comment (wlindner) [reply starters/blog/src/components/seo.js]: Ok, good to know. Thanks! I just committed a correction.

**Comment by gustavpursche** [comment]
@wlindner I hadn't heart about the issue before and when I checked yesterday, I saw recent commints in [react-helmet](https://github.com/nfl/react-helmet). Seems like they are picking the open issues up (again).

**Comment by wlindner** [comment]
Hmm, I see what you mean (https://github.com/nfl/react-helmet/releases) looks like they've created a new release recently after a long hiatus. Maybe it's ok to keep things the way they are. I saw this issue thread https://github.com/nfl/react-helmet/issues/465 and decided to move my project (that started from `gatsby-starter-blog`) to `react-helmet-async` but now that you mention it, a maintainer of `react-helmet` responded to that issue and might address the warning that users see when they use `react-helmet`. So... I guess, we don't really need this change.

**Comment by wardpeet** [comment]
Thank you for creating this Pull Request, it makes sense to move to another package but for now we suggest people to use the beta packages, we haven't seen any issues with that so far. Lots of our docs refer to `gatsby-plugin-react-helmet`. (it seems like we will get a 6.x release soon (https://github.com/nfl/react-helmet/pull/472#issuecomment-578600310).

Thanks again, and we look forward to seeing more PRs from you in the future! 💪💜

**Comment by JustFly1984** [comment]
@wardpeet @blainekasten please update react-helmet@6.0.0 https://github.com/nfl/react-helmet/issues/426#issuecomment-611716119

**Comment by wardpeet** [comment]
Sweet @JustFly1984 thanks for letting us know.

**Pull request closed without merging**